src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -162,43 +256,56 @@ func (sm *StorageMinerActor_I) SubmitPoSt(postSubmission poster.PoStSubmission)
 			panic("Invalid sector state in ProvingSet.SectorsOn()")
 		}
 	}
-	// power update here, because expiraiton updates again.
-	// SendMessage(sma.UpdatePower(powerUpdate))
+
+	// commit state change so that committed and recovering are now active
 
 	// Process ProvingSet.SectorsOff()
 	// ProvingSet.SectorsOff() contains SectorFailing
 	// SectorRecovering is Proving and hence will not be in GetZeros()
 	// heavy penalty if Failing for more than or equal to MAX_CONSECUTIVE_FAULTS
-	// otherwise increment FaultCount in SectorStates()
+	// otherwise increment FaultCount in Sectors().State
 	for _, sectorNo := range sm.ProvingSet_.SectorsOff() {
-		state := sm.SectorStates()[sectorNo]
-		switch state.StateNumber {
+		state, found := sm.Sectors()[sectorNo]
+		if !found {
```

Comment:
note why -- `// sm.ProvingSet_.SectorsOff() includes both failing and some cleared sectors -- only want failing`

Reply:
cc @whyrusleeping